crates/swc_plugin_runner/src/runtime.rs

```diff
@@ -0,0 +1,131 @@
+//! Plugin runtime abstract
+
+use std::{any::Any, fmt, path::Path};
+
+/// Runtime FFI Value
+pub type Value = i32;
+
+/// Opaque module cache
+pub struct ModuleCache(pub Box<dyn Any + Send + Sync>);
```

Comment:
nit: I think it can be better if we do not expose `Any` as a public API as it allows user to specify anything, although it would be caught in runtime

Reply:
It must be public so that users can access the custom type.

We could use a custom trait instead, but this would make the downcast unsafe and still not avoid runtime check.